matchzoo/preprocessors.py

```diff
@@ -1,146 +1,165 @@
+    def transform(self, input: str) -> list:
+        """
+        Process input data from raw terms to list of tokens.
 
-    :return tokens: list of segmentated tokens.
-    """
-    return list(jieba.cut(raw))
+        :param input: raw textual input.
 
+        :return tokens: tokenized tokens as a list.
+        """
+        return nltk.word_tokenize(input)
 
-def to_lowercase(tokens: list) -> list:
-    """
-    Convert list of tokens to lower case.
 
-    :param tokens: list of tokens.
+class LowercaseUnit(StatelessProcessUnit):
+    """Process unit for text lower case."""
 
-    :return tokens: lower-cased list of tokens.
-    """
-    return [token.lower() for token in tokens]
+    def transform(self, tokens: list) -> list:
+        """
+        Convert list of tokens to lower case.
 
+        :param tokens: list of tokens.
 
-def remove_stopwords(tokens: list, lang: str='en') -> list:
-    """
-    Remove stopwords from list of tokenized tokens.
+        :return tokens: lower-cased list of tokens.
+        """
+        return [token.lower() for token in tokens]
 
-    :param tokens: list of tokenized tokens.
-    :param lang: language code for stopwords.
 
-    :return tokens: list of tokenized tokens without stopwords.
-    """
-    return [token for token in tokens if token not in get_stopwords(lang)]
+class PuncRemovalUnit(StatelessProcessUnit):
+    """Process unit for remove punctuations."""
 
+    def transform(self, tokens: list) -> list:
+        """
+        Remove punctuations from list of tokens.
 
-def remove_punctuation(tokens: list) -> list:
-    """
-    Remove punctuations from list of tokens.
+        :param tokens: list of toekns.
 
-    :param tokens: list of
+        :return rv: tokens  without punctuation.
+        """
+        return [token for token in tokens if not match_punc.search(token)]
 
-    :return rv: tokens  without punctuation.
-    """
-    return [token for token in tokens if not match_punc.search(token)]
 
+class DigitRemovalUnit(StatelessProcessUnit):
+    """Process unit to remove digits."""
 
-def remove_digits(tokens: list) -> list:
-    """
-    Remove digits from list of tokens.
+    def transform(self, tokens: list) -> list:
+        """
+        Remove digits from list of tokens.
 
-    :param tokens: list of tokens to be filtered.
+        :param tokens: list of tokens to be filtered.
 
-    :return tokens: tokens of tokens without digits.
-    """
-    return [token for token in tokens if not token.isdigit()]
+        :return tokens: tokens of tokens without digits.
+        """
+        return [token for token in tokens if not token.isdigit()]
 
 
-def stemming(tokens: list, stemmer: str = 'porter') -> list:
-    """
-    Reducing inflected words to their word stem, base or root form.
+class StopRemovalUnit(StatelessProcessUnit):
+    """Process unit to remove stop words."""
 
-    :param tokens: list of string to be stemmed.
+    def transform(self, tokens: list, **kwargs: dict) -> list:
```

Comment:
I suggest having a consistent `list` -> `list` signature for these type of processing units. Configuration properties like `language` can be set during object construction, i.e. in `__init__`.

Reply:
okay